src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -173,47 +174,66 @@ public override int GetMatsOsPlatformCode()
+                    var psi = new ProcessStartInfo
+                    {
+                        FileName = url,
+                        UseShellExecute = true
+                    };
+                    Process.Start(psi);
+                }
+                catch
                 {
+                    // hack because of this: https://github.com/dotnet/corefx/issues/10361
                     url = url.Replace("&", "^&");
                     Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                try
                 {
-                    try
+                    ProcessStartInfo psi = null;
+                    foreach (string openTool in new[] { "xdg-open", "gnome-open", "kfmclient" })
                     {
-                        Process.Start("xdg-open", url);
+                        if (TryGetExecutablePath(openTool, out string openToolPath))
+                        {
+                            psi = new ProcessStartInfo(openToolPath, url)
+                            {
+                                RedirectStandardOutput = true,
+                                RedirectStandardError = true
+                            };
+
+                            Process.Start(psi);
+
+                            break;
+                        }
                     }
-                    catch (Exception ex)
+
+                    if (psi == null)
                     {
-                        throw new MsalClientException(
-                            MsalError.LinuxXdgOpen,
-                            "Unable to open a web page using xdg-open. See inner exception for details. Possible causes for this error are: xdg-open is not installed or " +
-                            "it cannot find a way to open an url - make sure you can open a web page by invoking from a terminal: xdg-open https://www.bing.com ",
-                            ex);
+                        throw new Exception("Failed to locate a utility to launch the default web browser.");
                     }
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                catch (Exception ex)
                 {
-                    Process.Start("open", url);
-                }
-                else
-                {
-                    throw new PlatformNotSupportedException(RuntimeInformation.OSDescription);
+                    throw new MsalClientException(
+                        MsalError.LinuxXdgOpen,
+                        MsalErrorMessage.LinuxOpenToolFailed,
+                        ex);
                 }
             }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                Process.Start("open", url);
```

Comment:
I know the macOS `open` line here was in the existing code, but it might be good to also "fix" up this too whilst you're in here. Not specifying the full path to `open` has the same possible issue as before (the BCL will search for `./open` first, before the path).

In the runtime, they [hardcode the `open` path to be `/usr/bin/open`](https://github.com/dotnet/runtime/blob/82c705102a34c36c5395eb8804109f54c9750d90/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs#L45-L48):

```csharp
private string GetPathToOpenFile()
{
    return "/usr/bin/open";
}
```

Reply:
Updated. Thanks for the suggestion.